app/src/androidTest/java/com/example/warriorsocial/ui/login/RegistrationCheckerTest.java

```diff
@@ -0,0 +1,117 @@
+package com.example.warriorsocial.ui.login;
+
+import android.content.Context;
+import android.widget.EditText;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.auth.FirebaseAuth;
+
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class RegistrationCheckerTest {
+
+    private static RegistrationChecker registrationChecker;
+    private static User testUser;
+    private static FirebaseAuth fAuth;
+    private static Context context;
+
+    private static EditText testEmail;
+    private static EditText testEmail2;
+    private static EditText testEmail3;
+
+    private static EditText testPassword;
+    private static EditText testPassword2;
+    private static EditText testPassword3;
+    private static EditText testPassword4;
+
+    @BeforeClass
+    public static void before() {
+        // Initiate a context; In this case we need to create a "sample" context for unit testing using InstrumentationRegistry
+        // This is necessary as for one of the methods tested, context is required
+        context = InstrumentationRegistry.getInstrumentation().getTargetContext();
+        FirebaseApp.initializeApp(context);
+
+
+        // Initialize registrationChecker();
+        registrationChecker = new RegistrationChecker();
+
+        // Initialize sample EditTexts
+        testEmail = new EditText(context);
+        testEmail.setText("not@empty");
+        testEmail2 = new EditText(context);
+        testEmail2.setText("something@wayne");
+        testEmail3 = new EditText(context);
+        // no setText for testEmail3 ( should be empty )
+
+        testPassword = new EditText(context);
+        testPassword.setText("testing12321");
+        testPassword2 = new EditText(context);
+        testPassword2.setText("alex");
+        testPassword3 = new EditText(context);
+        // no setText for testPassword3 ( should be empty )
+        testPassword4 = new EditText(context);
+        testPassword4.setText("alex5");
+        // Initiate testUser for unit tests
+        testUser = new User("something", testEmail.getText().toString());
+
+        // Initialize Firebase Auth (Simulate a login)
+        fAuth = FirebaseAuth.getInstance();
+
+        fAuth.signInWithEmailAndPassword(testEmail.getText().toString(), testPassword.getText().toString());
+    }
+
+    @Test
+    public void checkIfUserLoggedIn() {
```

Comment:
Unit tests look great! The comments make it a lot easier to understand the test.